src/actions/users.js

```diff
@@ -1554,6 +1554,53 @@ export function clearUserAccessTokens() {
     };
 }
 
+/**
+ * Send a getUserAnalytics request for a user (userId) and team id (teamId)
+ * Pass a string indicating which dataset to return (name)
+ * user had in a particular team.
+ *
+ * Can potentially dispatch 4 action types
+ *     1. GET_USER_ANALYTICS_REQUEST
+ *        - sets requests.users.getUserAnalytics[name].status to 'started'
+ *     2. GET_USER_ANALYTICS_FAILURE
+ *        - sets requests.users.getUserAnalytics[name].status to 'failure'
+ *        - sets requests.users.getUserAnalytics[name].error to error from server
+ *     3. GET_USER_ANALYTICS_SUCCESS
+ *        - sets requests.users.getUserAnalytics[name].status to 'success'
+ *     4. RECEIVED_USER_ANALYTICS
+ *        - adds user analytics data returned to entitites.users.analytics
+ *
+ * @param {string} userId - The target user for the requested analytics
+ * @param {string} teamId - The target team for the requested analytics
+ * @param {string} name - The name of the type of analytic data to return
+ *
+ * @returns {Promise<object>} containing the particular analytics requested
```

Comment:
This is implemented like all of the other similar functions in this file, so I'm sure it's correct, but this function doesn't return a promise (it is not an async function, it returns an async function). That explicit type description is nuts to put here since we're not trying for perfect documentation.
The explicit type would be something like `{function(Function, Function): Promise<object>}` I think I would just call the type `AsyncThunk` which in the redux world implies the thunk is called w/ dispatch and getState functions as arguments, and because I named it 'async' implies it returns a promise.

All that said I would just change the return to be:
```
 * @returns {AsyncThunk<object>} that resolves with the requested analytics
```
Describing this return value is tough, but that's my best attempt.

Reply:
Ok I used your example. Yeah I forgot that this is a thunk that gets returned.